app/controllers/all-events.js

```diff
@@ -0,0 +1,320 @@
+            if (searchValue) {
+                let newList = [];
+                let oldList = this.get("model.allEvents");
+                console.log(oldList);
+                for (let i = 0; i < oldList.length; i++) {
+                    if (oldList[i].Event_Name.toLowerCase().indexOf(searchValue.toLowerCase()) !== -1) {
+                        newList.push(oldList[i]);
+                    } else if (oldList[i].End_Time.toLowerCase().indexOf(searchValue.toLowerCase()) !== -1) {
+                        newList.push(oldList[i]);
+                    } else if (oldList[i].Location !== null) {
+                        if (oldList[i].Location.toLowerCase().indexOf(searchValue.toLowerCase()) !== -1) {
+                            newList.push(oldList[i]);
+                        }
+                    } else if (oldList[i].Start_Time.toLowerCase().indexOf(searchValue.toLowerCase()) !== -1) {
+                        newList.push(oldList[i]);
+                    }
+
+                }
+                this.set("model.eventShown", newList);
+            } else {
+                this.set("model.eventShown", this.get("model.allEvents"));
+            }
+        },
+
+
+        toggleCheckBox: function() {
+
+            if (this.get("pastEvents")) {
+                this.set("model.eventShown", this.get("model.allEvents"));
+            } else {
+                this.set("model.eventShown", this.get("model.pastEvents"));
+            }
+
+            let newList = [];
+            let oldList = this.get("model.eventShown");
+
+
+            if (this.get("isChapel")) {
+                console.log(oldList);
+                for (let i = 0; i < oldList.length; i++) {
+                    if (oldList[i].Category_Id === "85") {
+                        newList.push(oldList[i]);
+                    }
+
+                }
+                this.set("model.eventShown", newList);
+            }
+            if (this.get("isAthletics")) {
+                console.log(oldList);
+                for (let i = 0; i < oldList.length; i++) {
+                    if (oldList[i].Organization === "Athletics") {
+                        newList.push(oldList[i]);
+                    }
+
+                }
+                this.set("model.eventShown", newList);
+            }
+            if (this.get("isAdmissions")) {
+                console.log(oldList);
+                for (let i = 0; i < oldList.length; i++) {
+                    if (oldList[i].Organization === "Admissions") {
+                        newList.push(oldList[i]);
+                    }
+
+                }
+                this.set("model.eventShown", newList);
+            }
+        },
+
+        showPastEvents(item) {
+            for (let i = 0; i < this.get("model.eventShown").length; i++) {
+                this.set("model.eventShown", this.get("model.allEvents"));
+                controller.toggleCheckBox();
+
+            }
+        },
+
+        toggleEventDetailsModal(item) {
```

Comment:
you might want to break up this code into a couple of pieces for easier readability.  You can have functions in the controllers just like in vanilla javascript.